entry.py

```diff
@@ -1,23 +1,53 @@
 
 from src import gpa
+from src.io import read_course_csv, read_student_csv
+from src.utils import write_result
 
+# import module
+from datetime import datetime
+
+# get current date and time
+# current_datetime = str(datetime.now())
+today = datetime.now()
+dt_string = today.strftime("%d%m%Y_%H_%M_%S")
+current_datetime = str(dt_string)
+
+gpa_file_name = "gpa_result_"+current_datetime
+cgpa_file_name = "cgpa_result_"+current_datetime
 
-student_records = {
-      "name": "James Webb",
-      "id": "APT5005",
-      "courses":[
-         {"title": "English", "unit": 2, "code": "ENG101", "score": 60, "term": "Fall", "session":'2020'},
-         {"title": "Chemistry I", "unit": 4, "code": "CHE101", "score": 70, "term": "Fall", "session":'2021'},
-         {"title": "Maths", "unit": 3, "code": "MTH101", "score": 80, "term": "Spring", "session":'2020'},
-         {"title": "Chemistry II", "unit": 4, "code": "CHE102", "score": 91, "term": "Spring", "session":'2021'}, 
-         {"title": "History", "unit": 2, "code": "HIS102", "score": 40, "term": "Spring", "session":'2020'}
-      ]
-   }
 
 
 if __name__ == "__main__":
-    results=gpa.compute_results(student_records)
-    print(results)
+   student_information = read_student_csv('data/student_record.csv')
+   course_information = read_course_csv('data/course_records.csv')
+
+   for course in course_information:
+      student_id = course["student_id"]
+
+      if student_id not in student_information:
+         print(f"student not found: {student_id}")
+         continue
+      student_information[student_id].setdefault("courses", []).append(course)
+
+overall_student_information = list(student_information.values())
+
+gpa_computed_result = []
+cgpa_computed_result = []
+for student in overall_student_information:
```

Comment:
```suggestion
for student_id, student in overall_student_information:
```
This way, you can pass the `student_id` around to relevant functions if required